src/plugins/kibana_usage_collection/server/collectors/common/counters.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import {
+  CollectorFetchContext,
+  UsageCountersSavedObject,
+  UsageCountersSavedObjectAttributes,
+} from '@kbn/usage-collection-plugin/server';
+
+export interface CounterEvent {
+  domainId: string;
+  counterName: string;
+  counterType: string;
+  lastUpdatedAt?: string;
+  fromTimestamp?: string;
+  total: number;
+}
+
+export function createCounterFetcher<T>(type: string, transform: (counters: CounterEvent[]) => T) {
+  return async ({ soClient }: CollectorFetchContext) => {
+    const finder = soClient.createPointInTimeFinder<UsageCountersSavedObjectAttributes>({
+      type,
+      namespaces: ['*'],
+      fields: ['count', 'counterName', 'counterType', 'domainId'],
+      perPage: 1000,
+    });
+
+    const dailyEvents: CounterEvent[] = [];
+
+    for await (const { saved_objects: rawUsageCounters } of finder.find()) {
+      rawUsageCounters.forEach((raw) => {
+        try {
+          const event = transformRawCounter(raw);
+          if (event) {
+            dailyEvents.push(event);
+          }
+        } catch (_) {
+          // swallow error; allows sending successfully transformed objects.
+        }
+      });
+    }
+
+    return transform(mergeCounters(dailyEvents));
+  };
+}
+
+export function transformRawCounter(
+  rawCounter: UsageCountersSavedObject
+): CounterEvent | undefined {
+  const {
+    attributes: { domainId, counterType, counterName, count },
+    updated_at: lastUpdatedAt,
+  } = rawCounter;
+
+  if (typeof count !== 'number' || count < 1) {
+    return;
+  }
+
+  const fromTimestamp = moment(lastUpdatedAt).utc().startOf('day').format();
```

Comment:
Mentioned in another comment, but just to point out in the code... `fromTimestamp` should probably be stored instead of calculated if we plan to query by it.

Reply:
As discussed, we can rely on `updated_at` attribute (part of the standard SO definition) for our filters.
I don't really know the purpose of these for Stack Telemetry (or why we need both `lastUpdatedAt` and `fromTimestamp`).